numpy/lib/shape_base.py

```diff
@@ -134,9 +134,9 @@ def apply_along_axis(func1d, axis, arr, *args, **kwargs):
         buff = res.__array_prepare__(buff)
 
     # save the first result, then compute and save all remaining results
-    buff[ind0] = res
+    buff[ind0] = res[()]
```

Comment:
I didn't realize that indexing like this was valid on non-scalar arrays. This needs a comment, or better yet, maybe a helper function. Something like `asanyarray_or_scalar`? This is related to #5353, though in this case we want fallback behavior.

Alternatively, maybe we should change indexing behavior for cases like this in general? I don't see any situation in which a user would want an object array of scalar object arrays.